sparse/_common.py

```diff
@@ -717,15 +718,15 @@ def _dot_csr_ndarray(out_shape, a_data, a_indices, a_indptr, b):  # pragma: no c
         out_shape : Tuple[int]
             The shape of the output array.
         """
-        out = np.empty(out_shape, dtype=dtr)
+        b = np.ascontiguousarray(b)  # ensure memory aligned
```

Comment:
If the array is not C- contiguous this will cause a copy to be made... Up to you all if that's okay, but it does noticeably speed up operations for those cases when it is not C-contiguous.

Reply:
For reference, `scipy` does this sort of thing with their sparse arrays, as the `ndarray`s are always raveled before the actual multiplication happens.

Reply:
Thanks for looking into this! Yes, I expect that these arrays will be contiguous most of the time and therefore this to be a net benefit. I would also make it contiguous when creating the arrays in the first place.